indexing-service/src/main/java/org/apache/druid/indexing/common/task/batch/parallel/iterator/RangePartitionIndexTaskInputRowIteratorBuilder.java

```diff
@@ -114,9 +114,18 @@ public HandlingInputRowIterator build()
   )
   {
     return inputRow -> {
-      List<String> dimensionValues = inputRow.getDimension(partitionDimension);
-      return dimensionValues.size() > 1;  // Rows.objectToStrings() returns an empty list for a single null value
+      int dimensionValueCount = getDimensionValueCount(inputRow, partitionDimension);
+      return dimensionValueCount > 1;  // Rows.objectToStrings() returns an empty list for a single null value
     };
   }
 
+  private static int getDimensionValueCount(InputRow inputRow, String partitionDimension)
```

Comment:
Nothing in this function indicates that it will throw an exception if `dimensionValueCount > 1`

Maybe rename the function to `getSingleOrNullDimensionValueCount` so that it's a little more obvious the scope of this function is very narrow. As written, I'd definitely use it just expecting it to return the number of dimensions